nionswift_plugin/usim/CameraDevice.py

```diff
@@ -218,19 +284,21 @@ def _acquire_sequence(self, n: int) -> typing.Optional[ImportExportManager.DataE
         # if the device does not implement acquire_sequence, fall back to looping acquisition.
         self.__is_acquiring = True
         self.__has_data_event.clear()  # ensure any has_data_event is new data
-        self.__instrument.sequence_progress = 0
+        self.__original_exposure = self.__exposure
         try:
             properties = None
             data = None
-            if self._external_trigger:
-                scan_device: typing.Optional[ScanDevice.Device] = Registry.get_component("scan_device")
-                if scan_device and hasattr(scan_device, "blanker_signal_condition"):
-                    with scan_device.blanker_signal_condition:
-                        scan_device.blanker_signal_condition.wait(timeout=max(self.__exposure * 2, 5))
             for index in range(n):
                 if self.__cancel_sequence_event.is_set():
                     return None
+                # Time the actual duration of a frame and adjust the internal exposure time accordingly to account for
+                # overhead. This is important when acquiring data where synchronization between scan and camera is
+                # achieved via having them run at the same speed.
+                frame_start_time = time.time()
                 frame_data_element = self.__acquire_image(direct=True)
+                frame_time = time.time() - frame_start_time
+                difference = frame_time - self.__original_exposure
+                self.__exposure -= difference
```

Comment:
The logic here (with the subtractions) is difficult to follow. It leads to negative `self.__exposure`, e.g. see `test_acquire_sequence_1d_calibrations` where exposure is set to 0.04s and elapsed frame time takes 0.25s. This is in a loop, so `self.__exposure` may keep getting smaller and smaller.

Also, modifying `self.__exposure` and restoring it is essentially subverting the meaning of `self.__exposure` and making it act as a global variable. I think we should find a different technique of doing this - perhaps pass the exposure to `__acquire_image` and subsequently `__direct_acquire` (where `self.__exposure` is used).

Overall I don't like making the exposure be affected by a measured value. Already some tests fail and I think it's likely to have similar issues in the future.

One brainstorming idea is to define some sort of "simulator system clock" and base the simulator behavior on that system clock. By default it would just be "immediate" but in the case of burst mode we could gate all of the synchronization on that common clock object.

Reply:
That's a good point, I should come up with a better solution. The common clock object sounds good for internal use, but it would make using other camera simulators that are not included in nionswift-usim more difficult to use. I'd really like to make the timing in our scan and camera simulators as accurate as possible so that we can do synchronization in a way that is as close to how it works with real hardware as possible. Of course we are pretty limited by the timing accuracy of the os with this, so maybe we still need to go for some common clock object.

Reply:
Fixed in https://github.com/nion-software/nionswift-usim/pull/36/commits/f5c321330420507ae3fd79a107d9439bc92f0c8e